rest/admin_api_test.go

```diff
@@ -4125,6 +4125,33 @@ func TestEmptyStringJavascriptFunctions(t *testing.T) {
 	assert.Equal(t, http.StatusCreated, resp.StatusCode)
 }
 
+// Regression test for CBG-2119 - ensure that bool fields are handled correctly both when set as true and as false
+func TestConfigResetBooleanFields(t *testing.T) {
+	rt := NewRestTester(t, nil)
+	defer rt.Close()
+
+	newConfig := map[string]interface{}{
+		"bucket":                      rt.Bucket().GetName(),
+		"num_index_replicas":          0,
+		"enable_shared_bucket_access": base.TestUseXattrs(),
+		"use_views":                   base.TestsDisableGSI(),
+		"disable_password_auth":       true,
+	}
+
+	newConfigJSON, err := base.JSONMarshal(newConfig)
+	require.NoError(t, err, "failed to marshal new config")
+	res := rt.SendAdminRequest(http.MethodPut, "/db/_config", string(newConfigJSON))
```

Comment:
It would be better to test the POST endpoint to confirm it upserts the `disable_password_auth` field correctly rather than replacing the whole config with the correct value which does not properly unit test this new change.